File: LeetCode/python3/524.py
class Solution:
    def findLongestWord(self, s: str, d: List[str]) -> str:
        res = ""
        for target in d:
            i, j = 0, 0
            while i < len(target) and j < len(s):
                if target[i] == s[j]:
                    i += 1
                j += 1
            if i == len(target):
                if len(target) > len(res) or (len(target) == len(res) and target < res):
                    res = target
        return res

File: LeetCode/cpp/524.cpp
class Solution
{
  public:
    string findLongestWord(string s, vector<string> &d) {
        string res = "";
        for (auto curr : d) {
            int j = 0, k = 0;
            while (k < s.size() && j < curr.size()) {
                if (s[k] == curr[j])
                    ++j;
                ++k;
            }
            if (j == curr.size())
                if (res.size() < curr.size() || (res.size() == curr.size() && res > curr))
                    res = curr;
        }
        return res;
    }
};
